src/core/thread/mle.cpp

```diff
@@ -3574,6 +3596,7 @@ void Mle::HandleChildUpdateResponse(RxInfo &aRxInfo)
         SetStateChild(GetRloc16());
 
         mRetrieveNewNetworkData = true;
+        mRequestRouteTlv        = true;
```

Comment:
Limit this behavior to FTDs?

Reply:
Good point! I added `#if OPENTHREAD_FTD` and `if (IsFullThreadDevice())` checks.